persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java

```diff
@@ -337,13 +328,62 @@ public void column(String column, Column.ColumnType type) {
     column(column, type, Column.Kind.Regular);
   }
 
+  public void as(String alias) {
+    if (functionCalls.isEmpty()) {
+      throw new IllegalStateException(
```

Comment:
Maybe there should be a test for this condition?

```java
    BuiltQuery<?> query =
        newBuilder()
            .select()
            .column("k2", "v1")
            .as("alias1") // call without aggregate
            .build();
```

Reply:
As for me, I do not view it as an error necessarily, but since we have `as` in the grammar, keeping the two argument method is probably not necessary and removing it will remove the possibility of confusion in this case.

Reply:
This code will not compile: 
```
      newBuilder()
            .select()
            .column("k2", "v1")
            .as("alias1") // call without aggregate
            .build();
```
So this it is not possible to call the `as` without adding a function call before. 
I've added this check to clearly state  that this situation should not occur.